app/controllers/user_controller.rb

```diff
@@ -799,4 +801,23 @@ def check_signup_allowed(email = nil)
 
     !blocked
   end
+
+  ##
+  # check if this user has a gravatar and set the user pref is true
+  def gravatar_enable(user)
+    # code from example https://en.gravatar.com/site/implement/images/ruby/
+    return if user.image.present?
+    hash = Digest::MD5.hexdigest(user.email.downcase)
+    url = "https://www.gravatar.com/avatar/#{hash}?d=404" # without d=404 we will always get an image back
+    response = OSM.http_client.get(URI.parse(url))
+    oldsetting = user.image_use_gravatar
+    user.image_use_gravatar = response.success?
+    if oldsetting != user.image_use_gravatar
+      flash[:warning] = if user.image_use_gravatar
```

Comment:
Just jogged my memory a bit and retested on my instance.

The check if a gravatar is available or not takes place when the user uses the link from the confirmation mail, that in itseld shows a notice that the e.mail has been changed. I suspect a work around would be to merge the messages / create two additional ones for the case when the gravatar enables status has changed.